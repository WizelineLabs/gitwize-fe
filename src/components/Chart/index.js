import React, {useEffect, useState, useRef} from 'react'
import { makeStyles, styled } from '@material-ui/core/styles'
import Grid from '@material-ui/core/Grid'
import { Bar, Line } from 'react-chartjs-2'

const useStyles = makeStyles((theme) => ({
  root: {
    borderRadius: 4,
    padding: '30px 40px 30px 10px',
    height: '50vh',
    maxHeight: '50vh'
  },
}))

export const chartTypeEnum = {
  LINE: 'line',
  BAR: 'bar',
}

const ChartLegend = styled('div')(({
  theme
}) => ({
  "& li span": {
    width: '40px',
    height: '12px',
    display: 'inline-block',
    margin: '0 5px 8px 0',
    verticalAlign: '-9.4px'
  },
  "& ul": {
    display: 'flex',
    justifyContent: 'center',
    listStyle: 'none',
    font: '12px',
    whiteSpace: 'nowrap'
  },
  "& li": {
    cursor: 'pointer',
    textAlign: 'left',
    margin: '0px 25px',
    height: '20px',
    fontWeight: 'bold'
  }
}))

const buildChartBasedOnChartType = (chartType, chartRef, data, chartOptions, plugins) => {
  switch (chartType) {
    case chartTypeEnum.LINE:
      return <Line ref={chartRef} data={data} options={chartOptions} plugins={plugins} />
    case chartTypeEnum.BAR:
      return <Bar ref={chartRef} data={data} options={chartOptions} plugins={plugins} />
    default:
      return <Bar ref={chartRef} data={data} options={chartOptions} plugins={plugins} />
  }
}

export default function Chart(props) {

  const chartRef = useRef(null)
  const [legendCallbackGenerate, setLegendCallbackGenerate] = useState(false)
  const {data, chartOptions, customToolTip, customsStyle, customHandleClickLegend, customPlugins = [], isLegendClickable=true, chartLegendId = 'chart-legend', 
          chartType = chartTypeEnum.BAR, disableLegend = false} = props

  const classes = useStyles()

  const handleClick = (e, item, index, originalColor) => {
    let ci = chartRef.current.chartInstance;
    let meta = ci.getDatasetMeta(index);
    meta.hidden = meta.hidden === null ? !ci.data.datasets[index].hidden : null;
    if(item.style.color !== 'grey') {
      item.style.color = 'grey'
      item.childNodes[0].style.backgroundColor = 'grey'
      item.style.fontWeight = 'normal'
    } else {
      item.style.color = 'black'
      item.childNodes[0].style.backgroundColor = originalColor
      item.style.fontWeight = 'bold'
    }

    if(customHandleClickLegend) {
      customHandleClickLegend(ci, ci.data.datasets)
    }

    ci.update();
  }

  const newChartOptions = {
    ...chartOptions,
    responsive: true,
    elements: {
      line: {
        fill: false
      }
    },
    legend: {
      display: false
    },
    legendCallback: (chartInstance) => {
      let text = [];
      text.push('<ul>');
      for (let i = 0; i < chartInstance.data.datasets.length; i++) {
        text.push('<li><span style="background-color:' + chartInstance.data.datasets[i].backgroundColor + '"></span>');
        if (chartInstance.data.datasets[i].label) {
          text.push(chartInstance.data.datasets[i].label);
        }
        text.push('</li>');
      }
      text.push('</ul>');
      return text.join("");
    }
  }

  const plugins = [
  ...customPlugins,
  {
    afterDraw: (chartInstance) => {
      setLegendCallbackGenerate(true)
    }
  }]

  const buildCustomToolTip = (toolTipModel) => {
    customToolTip(toolTipModel, chartRef)
  }

  useEffect(() => {
    if(legendCallbackGenerate) {
      const generateLegend = () => {
        document.getElementById(
          chartLegendId
        ).innerHTML = chartRef.current.chartInstance.generateLegend();
  
        document.querySelectorAll(`#${chartLegendId} li`).forEach((item, index) => {
          const originalColor = item.childNodes[0].style.backgroundColor
          if(isLegendClickable) {
            item.addEventListener("click", e => handleClick(e, item, index, originalColor));
          }
          //keep color as grey if already disabled
          let ci = chartRef.current.chartInstance;
          const meta = ci.getDatasetMeta(index);
          if(meta.hidden) {
            item.style.color = 'grey'
            item.childNodes[0].style.backgroundColor = 'grey'
            item.style.fontWeight = 'normal'
          }
        });
      }
      if(!disableLegend) {
        generateLegend()
      }

      const newChartOptions = {
        ...chartOptions,
      }
      
      if(customToolTip) {
        newChartOptions.tooltips.custom = buildCustomToolTip
      }
      
      chartRef.current.chartInstance.options = {
        ...chartRef.current.chartInstance.options,
        ...newChartOptions
      }
      chartRef.current.chartInstance.update();
    }
  // eslint-disable-next-line
  }, [legendCallbackGenerate, chartOptions]);

  let chart;
  if(data && data.length !== 0) {
    chart = buildChartBasedOnChartType(chartType, chartRef, data, newChartOptions, plugins)
  } 

  return (
    <Grid container >
      <Grid className={classes.root} style={customsStyle} item xs={12}>
        {chart}
        <ChartLegend id={chartLegendId}/>
      </Grid>
    </Grid>
  )
}
