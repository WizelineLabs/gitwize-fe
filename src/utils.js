import Papa from 'papaparse'

export const formatData = data => {

  //console.log(data)
  let inputData = []
  data.map((row, i) => {
    const country = row.Country || null,
    year = row.Year || null,
    male = row[" Female"] || null,
    female = row[" Male"] || null

    if(!inputData[year]) inputData[year] = {}
    if(year) inputData[year].year = year
    if(country) inputData[year][country] = (male + female) / 2

  })


  let i = 0
  let finalData = []
  inputData.map(row => {
    finalData[i] = row
    ++i
  })

  return finalData
}

export const readDataFromFile = filePath => {

  return new Promise((resolve, reject) => {
    Papa.parse(filePath, {
      download: true,
      header: true,
      dynamicTyping: true,
      complete: function(result) {
          resolve(result.data)
      }
    })

  })

}





