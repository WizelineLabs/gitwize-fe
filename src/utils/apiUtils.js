export const convertDateToSecond = (date) => {
  return Math.floor(date.getTime()/1000)
}


export const calculateChangePercent = (additions, deletions, loc) => {
  return loc ? (((additions + deletions) / loc) * 100).toFixed(2) : 0
}

export const transformMetricsDataApiResponse = (data, dateRange) => {
  let { date_from, date_to } = dateRange
  date_from = convertDateToSecond(date_from)
  date_to = convertDateToSecond(date_to)
  const dataByDate = {}

  Object.keys(data).forEach((metric) => {
    if(data[metric] !== null) {
      data[metric].forEach(({ as_of_date: date, value }) => {
        dataByDate[date] = dataByDate[date]
          ? { ...dataByDate[date], [metric]: value }
          : { [metric]: value }
      })
    }
  })

  const dateList = []
  while(date_from <= date_to) {
    const date = (new Date(date_from*1000)).toLocaleDateString()
    
    const metricByDate = dataByDate[date] ? dataByDate[date] : {}

    dateList.push({
      Date: date,
      Commits: metricByDate.commits ? metricByDate.commits : 0,
      Additions: metricByDate.lines_added !== undefined ? metricByDate.lines_added : 0,
      Deletions: metricByDate.lines_removed !== undefined ? metricByDate.lines_removed : 0,
      Open: metricByDate.prs_created !== undefined ? metricByDate.prs_created : 0,
      Merged: metricByDate.prs_merged !== undefined ? metricByDate.prs_merged : 0,
      Rejected:metricByDate.prs_rejected !== undefined ? metricByDate.prs_rejected : 0
    })

    date_from = date_from + (24*3600)
  }

  return dateList
}

export const getRepositoryNameFromGitHubUrl = (url) => {
  const splittedUrl = url.split("/")
  const finalPart = splittedUrl[splittedUrl.length - 1]

  const repoName = finalPart.replace(".git", "")
  
  return repoName === null ? url : repoName
}
