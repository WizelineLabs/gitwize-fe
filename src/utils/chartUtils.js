import Papa from 'papaparse'
import {cloneDeep} from 'lodash'

export const readDataFromFile = (filePath) => {
  return new Promise((resolve) => {
    Papa.parse(filePath, {
      download: true,
      header: true,
      dynamicTyping: true,
      complete(result) {
        resolve(result.data)
      },
    })
  })
}

export const getChartColor = (data) => {
  switch (data) {
    case 'Commits':
    case 'Merged':
      return '#000000'
    case 'Additions':
    case 'Rejected':
      return '#00000052'
    case 'Deletions':
    case 'Created':
      return '#0000008f'
    case 'Pull requests':
      return '#0000008f'
    case 'Total lines of code':
      return '#00000052'
    default:
      return '#000000'
  }
}

export const getChartOptions = (chartOptions, chartLines = []) => {
  let newChartOptions = cloneDeep(chartOptions)
  if(chartLines.length > 0 && newChartOptions) {
    chartLines.forEach(item => {
      const yAxisItem = {
        type: 'linear',
        display: true,
        position: 'right',
        id: item.yAxisId,
        gridLines: {
          display: false
        },
        labels: {
          show: true
        },
        stacked: true,
        ticks: {
          fontColor: item.color,
          fontSize: 10,
          beginAtZero: true,
          precision: 0,
          suggestedMax: 10
        }
      }
      newChartOptions.scales.yAxes.push(yAxisItem);
    })
  }
  return newChartOptions
}

export const buildChartOptionsBasedOnMaxValue = (chartData) => {
  if(chartData) {
    const chartValue = Object.values(chartData)
    let maxValue = 0
    if(chartValue && chartValue.length !== 0) {
      maxValue = chartValue.reduce((a, b) => {
        return Math.max(a, b);
      })
    }
    return  {
      scales: {
        xAxes: [
          {
            display: true,
            gridLines: {
              display: true,
              drawTicks: false,
              drawOnChartArea: false,
            },
            stacked: false,
            ticks: {
              fontColor: "#C4C4C4",
              fontSize: 10,
              autoSkip: true,
              autoSkipPadding: 30
            }
          }
        ],
        yAxes: [
          {
            type: 'linear',
            display: false,
            position: 'left',
            id: 'y-axis-1',
            gridLines: {
              display: false
            },
            labels: {
              show: false
            },
            stacked: false,
            ticks: {
              fontColor: "#C4C4C4",
              fontSize: 10,
              beginAtZero: true,
              min: 0,
              max: (maxValue + (maxValue/2)),
              precision: 0,
              suggestedMax: 5
            }
          }
        ]
      },
      tooltips: {
        enabled: false
      },
      plugins: {
        datalabels: {
            anchor: 'end',
            align: 'right',
            offset: 11,
            font: {
              size: 13
            }
        }
      },
      maintainAspectRatio: false,
    }
  }
  return null
}
