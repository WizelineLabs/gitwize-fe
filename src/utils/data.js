export const data = {
  "loc": [
    {
      "branch": "master",
      "value": 2000,
      "asOfDate": "12/10/2020"
    },
    {
      "branch": "master",
      "value": 2301,
      "asOfDate": "13/10/2020"
    },
    {
      "branch": "master",
      "value": 1202,
      "asOfDate": "14/10/2020"
    },
    {
      "branch": "master",
      "value": 1003,
      "asOfDate": "15/10/2020"
    },
    {
      "branch": "master",
      "value": 1464,
      "asOfDate": "16/10/2020"
    },
    {
      "branch": "master",
      "value": 2125,
      "asOfDate": "17/10/2020"
    },
    {
      "branch": "master",
      "value": 2925,
      "asOfDate": "18/10/2020"
    },
    {
      "branch": "master",
      "value": 1553,
      "asOfDate": "19/10/2020"
    },
    {
      "branch": "master",
      "value": 1663,
      "asOfDate": "20/10/2020"
    }
  ],
  "lines_added": [
    {
      "branch": "master",
      "value": 1000,
      "asOfDate": "12/10/2020"
    },
    {
      "branch": "master",
      "value": 1300,
      "asOfDate": "13/10/2020"
    },
    {
      "branch": "master",
      "value": 2300,
      "asOfDate": "14/10/2020"
    },
    {
      "branch": "master",
      "value": 2050,
      "asOfDate": "15/10/2020"
    },
    {
      "branch": "master",
      "value": 1032,
      "asOfDate": "16/10/2020"
    },
    {
      "branch": "master",
      "value": 1302,
      "asOfDate": "17/10/2020"
    },
    {
      "branch": "master",
      "value": 1201,
      "asOfDate": "18/10/2020"
    },
    {
      "branch": "master",
      "value": 1111,
      "asOfDate": "19/10/2020"
    },
    {
      "branch": "master",
      "value": 2222,
      "asOfDate": "20/10/2020"
    }
  ],
  "lines_removed": [
    {
      "branch": "master",
      "value": 1000,
      "asOfDate": "12/10/2020"
    },
    {
      "branch": "master",
      "value": 1300,
      "asOfDate": "13/10/2020"
    },
    {
      "branch": "master",
      "value": 2300,
      "asOfDate": "14/10/2020"
    },
    {
      "branch": "master",
      "value": 2050,
      "asOfDate": "15/10/2020"
    },
    {
      "branch": "master",
      "value": 1032,
      "asOfDate": "16/10/2020"
    },
    {
      "branch": "master",
      "value": 1302,
      "asOfDate": "17/10/2020"
    },
    {
      "branch": "master",
      "value": 1201,
      "asOfDate": "18/10/2020"
    },
    {
      "branch": "master",
      "value": 1111,
      "asOfDate": "19/10/2020"
    },
    {
      "branch": "master",
      "value": 2222,
      "asOfDate": "20/10/2020"
    }
  ],
  "commits": [
    {
      "branch": "master",
      "value": 1234,
      "asOfDate": "12/10/2020"
    },
    {
      "branch": "master",
      "value": 1302,
      "asOfDate": "13/10/2020"
    },
    {
      "branch": "master",
      "value": 2400,
      "asOfDate": "14/10/2020"
    },
    {
      "branch": "master",
      "value": 1050,
      "asOfDate": "15/10/2020"
    },
    {
      "branch": "master",
      "value": 1232,
      "asOfDate": "16/10/2020"
    },
    {
      "branch": "master",
      "value": 1802,
      "asOfDate": "17/10/2020"
    },
    {
      "branch": "master",
      "value": 1601,
      "asOfDate": "18/10/2020"
    },
    {
      "branch": "master",
      "value": 1411,
      "asOfDate": "19/10/2020"
    },
    {
      "branch": "master",
      "value": 2322,
      "asOfDate": "20/10/2020"
    }
  ],
  "prs_created": [
    {
      "branch": "master",
      "value": 1304,
      "asOfDate": "12/10/2020"
    },
    {
      "branch": "master",
      "value": 2700,
      "asOfDate": "13/10/2020"
    },
    {
      "branch": "master",
      "value": 2760,
      "asOfDate": "14/10/2020"
    },
    {
      "branch": "master",
      "value": 2620,
      "asOfDate": "15/10/2020"
    },
    {
      "branch": "master",
      "value": 4431,
      "asOfDate": "16/10/2020"
    },
    {
      "branch": "master",
      "value": 3332,
      "asOfDate": "17/10/2020"
    },
    {
      "branch": "master",
      "value": 1189,
      "asOfDate": "18/10/2020"
    },
    {
      "branch": "master",
      "value": 2511,
      "asOfDate": "19/10/2020"
    },
    {
      "branch": "master",
      "value": 2221,
      "asOfDate": "20/10/2020"
    }
  ],
  "prs_merged": [
    {
      "branch": "master",
      "value": 1200,
      "asOfDate": "12/10/2020"
    },
    {
      "branch": "master",
      "value": 1800,
      "asOfDate": "13/10/2020"
    },
    {
      "branch": "master",
      "value": 2700,
      "asOfDate": "14/10/2020"
    },
    {
      "branch": "master",
      "value": 2650,
      "asOfDate": "15/10/2020"
    },
    {
      "branch": "master",
      "value": 1432,
      "asOfDate": "16/10/2020"
    },
    {
      "branch": "master",
      "value": 1302,
      "asOfDate": "17/10/2020"
    },
    {
      "branch": "master",
      "value": 1209,
      "asOfDate": "18/10/2020"
    },
    {
      "branch": "master",
      "value": 1181,
      "asOfDate": "19/10/2020"
    },
    {
      "branch": "master",
      "value": 2221,
      "asOfDate": "20/10/2020"
    }
  ],
  "prs_rejected": [
    {
      "branch": "master",
      "value": 1200,
      "asOfDate": "12/10/2020"
    },
    {
      "branch": "master",
      "value": 1800,
      "asOfDate": "13/10/2020"
    },
    {
      "branch": "master",
      "value": 2700,
      "asOfDate": "14/10/2020"
    },
    {
      "branch": "master",
      "value": 2650,
      "asOfDate": "15/10/2020"
    },
    {
      "branch": "master",
      "value": 1432,
      "asOfDate": "16/10/2020"
    },
    {
      "branch": "master",
      "value": 1302,
      "asOfDate": "17/10/2020"
    },
    {
      "branch": "master",
      "value": 1209,
      "asOfDate": "18/10/2020"
    },
    {
      "branch": "master",
      "value": 1181,
      "asOfDate": "19/10/2020"
    },
    {
      "branch": "master",
      "value": 2221,
      "asOfDate": "20/10/2020"
    }
  ]
}