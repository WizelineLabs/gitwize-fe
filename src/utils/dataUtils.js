import * as moment from 'moment'

const dateFormat = 'Do MMM'

export const createReversedArray = (array) => {
  const reversedArray = array.reduce((acc, b) => [b, ...acc], [])

  return reversedArray
}

export const transformPeriodToDateRange = (period) => {
  const today =  new Date()
  let endDay = null
  if(period === 'Last 7 Days')
    endDay = new Date(today.getTime() - (6 * 24 * 60 * 60 * 1000))
  else if(period === 'Last 14 Days')
    endDay = new Date(today.getTime() - (13 * 24 * 60 * 60 * 1000))
  else if(period === 'Last 21 Days')
    endDay = new Date(today.getTime() - (20 * 24 * 60 * 60 * 1000))
  else if(period === 'Last 30 Days')
    endDay = new Date(today.getTime() - (29 * 24 * 60 * 60 * 1000))

  return {
    period_date_from: endDay,
    period_date_to: today
  }
}

export const transformToChartData = (lines, bars, rawData, xAxis) => {
  let dataSets = [];
  let labels;
  if(xAxis === 'Date') {
    labels = rawData.flatMap(item => moment(item[xAxis]).format(dateFormat));
  } else {
    labels = rawData.flatMap(item => item[xAxis]);
  }
  lines.forEach(chartItem => {
    const dataArray = rawData.flatMap(rawDataItem => rawDataItem[chartItem.name]);
    let yAxisId = chartItem.yAxisId
    const dataSetsItem = {
      label: chartItem.name,
      type:'line',
      data: dataArray,
      fill: false,
      borderColor: chartItem.color,
      backgroundColor: chartItem.color,
      pointBorderColor: chartItem.color,
      pointBackgroundColor: chartItem.color,
      pointHoverBackgroundColor: chartItem.color,
      pointHoverBorderColor: chartItem.color,
      borderDash: chartItem.dash,
      yAxisID: yAxisId,
      lineTension: 0.001
    }
    dataSets.push(dataSetsItem);
  })
  bars.forEach(chartItem => {
    const dataArray = rawData.flatMap(rawDataItem => rawDataItem[chartItem.name]);
    const dataSetsItem = {
      label: chartItem.name,
      type:'bar',
      data: dataArray,
      fill: false,
      stack: chartItem.stackId,
      backgroundColor: chartItem.color,
      borderColor: 'white',
      hoverBackgroundColor: chartItem.color,
      hoverBorderColor: chartItem.color,
      yAxisID: 'y-axis-1',
      barPercentage: chartItem.barPercentage ? chartItem.barPercentage : 0.5,
      categoryPercentage: chartItem.categoryPercentage ? chartItem.categoryPercentage : 0.9,
      borderWidth: {
        top: 2,
        right: 0,
        bottom: 0,
        left: 0
      }
    }
    dataSets.push(dataSetsItem);
  })
  return {
    labels: labels,
    datasets: dataSets
  }
}

export const filterTableData =  (tableData = [], tableColumn) => {
  return tableData.map((item) => {
    if(item['Date']) {
      item['Date'] = moment(item['Date']).format(dateFormat)
    }
    return Object.assign(...tableColumn.map((object) => ({[object.text]: item[object.fieldName]})))
  })
}

export const convertTableObjectToTableColumn =  (tableObject) => {
  return tableObject.flatMap(item => ({
    title: item.text,
    field: item.text,
    type: (item.type) ? item.type : 'string',
    searchable: item.searchable ? true : false,
    cellStyle: item.cellStyle
  }))
}
