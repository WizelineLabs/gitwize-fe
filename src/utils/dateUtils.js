import * as moment from 'moment'

export const getCurrentDate = () => {
    return moment();
}

export const getStartOfMonth = (date) => {
    return moment(date).startOf('month')
}

export const getEndOfMonth = (date) => {
    return moment(date).endOf('month')
}

export const getNumberOfMonthBackward = (date, number) => {
    return moment(date).subtract(number, 'month')
}