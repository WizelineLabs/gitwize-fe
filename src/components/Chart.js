import React from 'react'
import { ComposedChart, Line, XAxis, YAxis, CartesianGrid, Tooltip, ResponsiveContainer, Bar, Legend } from 'recharts'
import PropTypes from 'prop-types'
import { getChartColor } from '../utils/chartUtils'

function Chart(props) {
  const { xAxis, data, lines, bars } = props

  return (
    <ResponsiveContainer width="100%" height={300}>
      <ComposedChart cx="50%" cy="50%" outerRadius="80%" data={data}>
        <XAxis dataKey={xAxis} />
        <YAxis />
        <CartesianGrid strokeDasharray="3 3" />
        <Legend />
        <Tooltip />

        {bars.map((bar) => (
          <Bar dataKey={bar} fill={getChartColor(bar)} barSize={10} isAnimationActive={false} key={bar} />
        ))}

        {lines.map((line, index) => (
          <Line
            connectNulls
            type="monotone"
            dataKey={line}
            isAnimationActive={false}
            stroke={getChartColor(line)}
            strokeDasharray={lines.length > 1 && index + 1 === lines.length ? '4 4' : ''}
            key={line}
          />
        ))}
      </ComposedChart>
    </ResponsiveContainer>
  )
}

Chart.propTypes = {
  data: PropTypes.instanceOf(Array).isRequired,
  xAxis: PropTypes.string.isRequired,
  lines: PropTypes.instanceOf(Array),
  bars: PropTypes.instanceOf(Array),
}

Chart.defaultProps = {
  lines: [],
  bars: [],
}

export default Chart
