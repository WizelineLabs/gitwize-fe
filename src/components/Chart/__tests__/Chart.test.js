import React from 'react'
import { shallow, mount } from 'enzyme'
import Chart from '../index'
import {transformToChartData} from '../../../utils/dataUtils'

const data = [
  {
    name: 'Page A',
    uv: 4000,
    pv: 2400,
    amt: 2400,
  },
  {
    name: 'Page B',
    uv: 3000,
    pv: 1398,
    amt: 2210,
  },
  {
    name: 'Page C',
    uv: 2000,
    pv: 9800,
    amt: 2290,
  },
  {
    name: 'Page D',
    uv: 2780,
    pv: 3908,
    amt: 2000,
  },
  {
    name: 'Page E',
    uv: 1890,
    pv: 4800,
    amt: 2181,
  },
  {
    name: 'Page F',
    uv: 2390,
    pv: 3800,
    amt: 2500,
  },
  {
    name: 'Page G',
    uv: 3490,
    pv: 4300,
    amt: 2100,
  },
]

describe('Chart', () => {
  it('render withour crashing', () => {
    const chartData = transformToChartData([{name: 'uv'}, {name: 'pv'}], [{name: 'amt'}], data, "name")
    const wrapper = mount(<Chart data={chartData}/>)
    expect(wrapper.find('Bar').length).toBe(1)
  })

  it('There should be a combination of bar and Line Chart', () => {
    const chartData = transformToChartData([{name: 'uv'}], [{name: 'amt'}], data, "name")
    const wrapper = mount(<Chart data={chartData} />)
    const bar = (wrapper.find('Bar'))
    const dataSets = bar.props().data.datasets
    expect(dataSets[0].type).toBe('line')
    expect(dataSets[1].type).toBe('bar')
  })

  test('There should be only 2 lines of Line Chart', () => {
    const chartData = transformToChartData([{name: 'uv'}, {name: 'pv'}], [], data, "name")
    const wrapper = mount(<Chart data={chartData}/>)
    const bar = (wrapper.find('Bar'))
    const dataSets = bar.props().data.datasets
    expect(dataSets[0].type).toBe('line')
    expect(dataSets[1].type).toBe('line')
  })

  test('There should be only bars of Bar Chart', () => {
    const chartData = transformToChartData([], [{name: 'amt'}], data, "name")
    const wrapper = mount(<Chart data={chartData}/>)
    const bar = (wrapper.find('Bar'))
    const dataSets = bar.props().data.datasets
    expect(dataSets[0].type).toBe('bar')
  })
})
