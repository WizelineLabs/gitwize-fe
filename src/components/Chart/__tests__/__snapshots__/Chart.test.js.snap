// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart should match snapshot renders 1`] = `ShallowWrapper {}`;
